inference-engine/src/cldnn_engine/cldnn_remote_context.h

```diff
@@ -204,6 +217,58 @@ class CLDNNRemoteAllocator : public InferenceEngine::IAllocator {
     bool free(void* handle) noexcept override { return true; }
 };
 
+class USMHostAllocator : public InferenceEngine::IAllocator {
+protected:
+    InferenceEngine::gpu::USMBlob::Ptr _usm_host_blob = nullptr;
+    InferenceEngine::gpu::ClContext* _context = nullptr;
+
+public:
+    using Ptr = std::shared_ptr<USMHostAllocator>;
+
+    USMHostAllocator(InferenceEngine::gpu::ClContext* context) : _context(context) { }
+    /**
+    * @brief Maps handle to heap memory accessible by any memory manipulation routines.
+    * @return Generic pointer to memory
+    */
+    void* lock(void* handle, InferenceEngine::LockOp = InferenceEngine::LOCK_FOR_WRITE) noexcept override {
+        if (!_usm_host_blob)
+            return nullptr;
+        return _usm_host_blob->get();
+    };
+
+    /**
+    * @brief Unmaps memory by handle with multiple sequential mappings of the same handle.
+    * The multiple sequential mappings of the same handle are suppose to get the same
+    * result while there isn't a ref counter supported.
+    */
+    void unlock(void* handle) noexcept override {}
+
+    /**
+    * @brief Allocates memory
+    * @param size The size in bytes to allocate
+    * @return Handle to the allocated resource
+    */
+    void* alloc(size_t size) noexcept override {
+        auto td = InferenceEngine::TensorDesc(InferenceEngine::Precision::U8, InferenceEngine::SizeVector{size}, InferenceEngine::Layout::C);
+        InferenceEngine::ParamMap params = {{GPU_PARAM_KEY(SHARED_MEM_TYPE), GPU_PARAM_VALUE(USM_HOST_BUFFER)}};
+        _usm_host_blob = std::dynamic_pointer_cast<InferenceEngine::gpu::USMBlob>(_context->CreateBlob(td, params));
+        _usm_host_blob->allocate();
+        return _usm_host_blob->get();
+    }
+
+    /**
+    * @brief Releases handle and all associated memory resources which invalidates the handle.
+    * @return false if handle cannot be released, otherwise - true.
+    */
+    bool free(void* handle) noexcept override {
+        try {
+            _usm_host_blob = nullptr;
+        } catch(...) { }
```

Comment:
The behavior is aligned with `SystemMemoryAllocator`. @ilya-lavrenov, @apankratovantonp why system allocator doesn't return false in case of exceptions in free method?

Reply:
Looks like it's a stupid behavior and `bool` means whether the memory was deallocated and user only for preallocator (where not actually memory is allocated / preallocated)